app/src/main/java/com/benefit/services/ChatService.java

```diff
@@ -0,0 +1,84 @@
+package com.benefit.services;
+
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.benefit.adapters.ConversationAdapter;
+import com.benefit.drivers.DatabaseDriver;
+import com.benefit.model.Chat;
+import com.benefit.model.Match;
+import com.benefit.model.User;
+import com.firebase.ui.firestore.FirestoreRecyclerOptions;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ChatService extends ViewModel {
+
+    private static final String TAG = ChatService.class.getSimpleName();
+
+    private DatabaseDriver databaseDriver;
+    private User user;
+    private CollectionReference matchCollectionRef;
+
+    public ChatService(){
+        this.databaseDriver = new DatabaseDriver();
+        matchCollectionRef = databaseDriver.getCollectionReferenceByName("matches");
+    }
+
+    public void setUser(User user){
+        if (user !=  null){
+            this.user = user;
+        }
+        else {
+            Log.d(TAG, "Error - user is null.");
+        }
+    }
+
+
+    public void addMatch(Match match){
+        matchCollectionRef.add(match);
+    }
+
+    public ConversationAdapter getConversationAdaptor(boolean wantedProducts, boolean userProducts, boolean orderNewToOld){
```

Comment:
is it type and should be getConversationAdapter?

Reply:
I change it to RecyclerView adpter